config/validate/validate.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+
+	json "github.com/ajeddeloh/go-json"
+	"github.com/coreos/ignition/config/types"
+	"github.com/coreos/ignition/config/validate/report"
+	"go4.org/errorutil"
+)
+
+type canValidate interface {
+	AssertValid() error
+}
+
+func canCallAssertValid(v reflect.Value) (canValidate, bool) {
+	val, ok := v.Interface().(canValidate)
```

Comment:
I think you want something closer to:

```go
if val, ok := v.Interface().(AssertValidator); ok {
    if v.Kind() == reflect.Ptr {
        val = v.Elem().Interface().(AssertValidator)
    }
    return val, true
} else {
    return nil, false
}
```

In the original code, if there is a pointer to something that has AssertValid(), it won't be called.